### PR TITLE
Add 4.15.1 release note

### PR DESCRIPTION
**Pull request by zymap**
---

*Motivation*

Generates the 4.15.1 docs and release note

### CODE REVIEW

**Comment by hangc0276** [comment]
The CI failed, https://github.com/apache/bookkeeper/runs/7672152565?check_suite_focus=true, please help take a look, thanks.

**Comment by StevenLuMT** [comment]
Dead link check failed: 
image

File: site3/website/src/pages/release-notes.md [hunk 1]
```diff
@@ -1,5 +1,69 @@
 # Release notes
 
+## 4.15.1
+
+Release 4.15.1 fixes some memories leak and do some improvements
+on stability and metrics.
+
+Apache BookKeeper users are encouraged to upgrade to 4.14.1.
```
Comment (StevenLuMT) [inline site3/website/src/pages/release-notes.md:8]: why 4.14.1? misstake?

**Comment by StevenLuMT** [comment]
rerun failure checks

**Comment by StevenLuMT** [comment]
fix old workflow,please see #3455 for detail

File: site3/website/src/pages/release-notes.md [hunk 2]
```diff
@@ -1,5 +1,69 @@
 # Release notes
 
+## 4.15.1
+
+Release 4.15.1 fixes some memories leak and do some improvements
+on stability and metrics.
```
Comment (dlg99) [inline site3/website/src/pages/release-notes.md:6]:
```suggestion
Release 4.15.1 contains performance and stability improvements, new metrics, and various bug fixes.
```

File: site3/website/src/pages/release-notes.md [hunk 3]
```diff
@@ -1,5 +1,69 @@
...
+
+Apache BookKeeper users are encouraged to upgrade to 4.14.1.
+The technical details of this release are summarized below.
+
+### Highlights
+
+#### Bugs
+
+* Fix autoRecover memory leak. https://github.com/apache/bookkeeper/pull/3361
```
Comment (dlg99) [inline site3/website/src/pages/release-notes.md:15]:
```suggestion
* Fix AutoRecovery memory leak. https://github.com/apache/bookkeeper/pull/3361
```

File: site3/website/src/pages/release-notes.md (hunk 1 above)
Comment (dlg99) [inline site3/website/src/pages/release-notes.md:8]:
```suggestion
Apache BookKeeper users are encouraged to upgrade to 4.15.1.
```

File: site3/website/src/pages/release-notes.md [hunk 4]
```diff
@@ -1,5 +1,69 @@
...
+* Ledger replicate supports throttle https://github.com/apache/bookkeeper/pull/2778
+* minorCompactionInterval should be greater than gcWaitTime https://github.com/apache/bookkeeper/pull/2116
+* add metric cbThreadPoolQueueSize https://github.com/apache/bookkeeper/pull/3424
+* Optimize concurrent collection's shrink logic https://github.com/apache/bookkeeper/pull/3417
+* Make BookieFileChannel interface public https://github.com/apache/bookkeeper/pull/3396
+* Prioritize compaction of entry logs with the lowest amount of remaining usable data https://github.com/apache/bookkeeper/pull/3390
+* Switch back ordered executor to LinkedBlockingQueue https://github.com/apache/bookkeeper/pull/3384
+* Consolidate Netty channel flushes to mitigate syscall overhead https://github.com/apache/bookkeeper/pull/3383
+* Shut down ReplicationWorker and Auditor on non-recoverable ZK error https://github.com/apache/bookkeeper/pull/3374
+* add journalCbQueueLatency https://github.com/apache/bookkeeper/pull/3364
```
Comment (dlg99) [inline site3/website/src/pages/release-notes.md:53]: I suggest moving this and other metrics-related PRs under separate section "Metrics changes" (same level as "Improvements") for better visibility

File: site3/website/src/pages/release-notes.md [hunk 5]
```diff
@@ -1,5 +1,69 @@
...
+* Fix the infinite waiting for shutdown due to throttler limit https://github.com/apache/bookkeeper/pull/2942
+* Make sure the LedgerHandle close callback can be completed when encounter exception https://github.com/apache/bookkeeper/pull/2913
+
+#### Improvements
+
+* If ensembleList is empty, return PlacementPolicyAdherence.FAIL. https://github.com/apache/bookkeeper/pull/3369
+* reduce unnecessary checkpoints https://github.com/apache/bookkeeper/pull/3341
+* Pre break loop when self create layoutZNode succeed. https://github.com/apache/bookkeeper/pull/3335
+* Avoid init WriteSet when waitForWriteSetMs < 0. https://github.com/apache/bookkeeper/pull/3325
+* Tuing PendingReadOp.java seq https://github.com/apache/bookkeeper/pull/3330
```
Comment (dlg99) [inline site3/website/src/pages/release-notes.md:35]:
```suggestion
* Tuning PendingReadOp.java seq https://github.com/apache/bookkeeper/pull/3330
```

**Review by dlg99 — CHANGES_REQUESTED** [review]
Overall lgtm, some suggestions related to spelling/wording.

Please consider formatting links to the PRs in the way that looks better on the website (`[PR #prnumber](https://url/to/pr)`)

**Comment by zymap** [comment]
I am fixing the CI issue

**Comment by zymap** [comment]
The dead link has already existed for a long time. Let's fix them with another PR. I opened an issue to track that.

**Review by dlg99 — APPROVED** [review]

**Pull request merged**